helpers/helper_nano.py

```diff
@@ -282,6 +282,216 @@ def ZMuMu_MuSelection(df):
+
+    df = df.Define('dr_mll', 'dR_mll(Electron_pt, Electron_eta, Electron_phi, isTag, isProbe)')
+
+    # Filter on pairs of lepton with DeltaR > 0.5
+    df = df.Filter('''
+    for (unsigned int line = 0; line < dr_mll.size(); line++){
+        for (unsigned int col = 0; col < 2; col++){
+            float DeltaR = dr_mll[line][col][0];
+            float mll = dr_mll[line][col][1];
+            if (mll > 60 && mll < 120){
+                return true;
+            }
+        }
+    }
+    return false;
+    ''')
+
+    # df = df.Define('_mll', 'mll(Electron_pt, Electron_eta, Electron_phi, isTag, isProbe)')
+    # df = df.Filter('_mll>60&&_mll<120')
+
+    df = df.Define('probe_Pt','Electron_pt[isProbe]')
+    df = df.Define('probe_Eta','Electron_eta[isProbe]')
+    df = df.Define('probe_Phi','Electron_phi[isProbe]')
+
+    return df
+
+
+def DQMOff_MuSelection(df):
+    '''
+    Selects Z->mumu events passing a single muon trigger. Defines probe pt/eta/phi
+    The selections are modified w.r.t ZMuMu_MuSelection method.
+    This is done so to match the DQM Off selections
+    '''
+    df = df.Filter('HLT_IsoMu27')
+
+    # Trigged on a Muon (probably redondant)
+    df = df.Filter('''
+    bool trigged_on_mu = false;
+    for (unsigned int i = 0; i < TrigObj_id.size(); i++){
+        if(TrigObj_id[i] == 13) trigged_on_mu = true;
+    }
+    return trigged_on_mu;
+    ''')
+
+    # TrigObj matching
+    df = df.Define('Muon_trig_idx', 'MatchObjToTrig_DQMOff(Muon_eta, Muon_phi, TrigObj_pt, TrigObj_eta, TrigObj_phi, TrigObj_id, 13, TrigObj_filterBits, 3, 0.1)')
+    df = df.Define('Muon_passHLT_IsoMu27', 'trig_is_filterbit_set_DQMOff(Muon_trig_idx, TrigObj_filterBits, 3)')
+    df = df.Define('Muon_PassTightId','Muon_tightId') 
+
+    df = df.Define('isTag','Muon_pt>26&&abs(Muon_pdgId)==13&&Muon_PassTightId&&Muon_passHLT_IsoMu27')
+    df = df.Filter('Sum(isTag)>0')
+
+    df = df.Define('isProbe','abs(Muon_pdgId)==13&&Muon_PassTightId&& (Sum(isTag)>=2||isTag==0)')
+    df = df.Define('dr_mll', 'dR_mll(Muon_pt, Muon_eta, Muon_phi, isTag, isProbe)')
+
+    # Filter on pairs of lepton with DeltaR > 0.5
+    df = df.Filter('''
+    for (unsigned int line = 0; line < dr_mll.size(); line++){
+        for (unsigned int col = 0; col < 2; col++){
+            float DeltaR = dr_mll[line][col][0];
+            if (DeltaR > 0.5){
+                return true;
+            }
+        }
+    }
+    return false;
+    ''')
+
+    df = df.Define('probe_Pt','Muon_pt[isProbe]')
+    df = df.Define('probe_Eta','Muon_eta[isProbe]')
+    df = df.Define('probe_Phi','Muon_phi[isProbe]')
+    df = df.Define('probe_Charge', 'Muon_charge[isProbe]')
+
+    return df
+
+
+def DQMOff_TauSelection(df):
+    '''
+    Selects Z->tautau events passing a single muon trigger. Defines probe tau pt/eta/phi
+    The selections are made to match with DQM Off selections
+    '''
+    df = df.Filter('HLT_IsoMu20')
```

Comment:
`HLT_IsoMu20` is heavily prescaled due to its L1 seed (`SingleMu18`). Please use `HLT_IsoMu24`, or better, `HLT_IsoMu24||HLT_IsoMu27`.

Reply:
I have changed it as suggested.